nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -277,11 +283,14 @@ WaypointFollower::followWaypoints()
       // Update server state
       goal_index++;
       new_goal = true;
+      result->wp_elapsed_time.push_back(now_time - last_wp_time);
+      last_wp_time = now_time;
       if (goal_index >= goal->poses.size()) {
         if (current_loop_no == no_of_loops) {
           RCLCPP_INFO(
-            get_logger(), "Completed all %zu waypoints requested.",
-            goal->poses.size());
+            get_logger(), "Completed all %zu waypoints requested in a time of %.2f seconds.",
+            goal->poses.size(),
+            elapsed_time.seconds());
```

Comment:
You can reference the feedback message here

Reply:
The message has a field called "sec" https://docs.ros2.org/foxy/api/builtin_interfaces/msg/Duration.html :wink: